File: unempty-desk-ui/src/utils/denormalizeRect.ts
import { Rect } from "@unempty-desk-ui/types/Rect";

export function denormalizeRect(normalized: Rect, width: number, height: number): Rect {
  return {
    x: Math.round(normalized.x * width),
    y: Math.round(normalized.y * height),
    width: Math.round(normalized.width * width),
    height: Math.round(normalized.height * height),
  };
}

File: unempty-desk-1/utils/denormalize_rect.py
from utils.constants import VIDEO_WIDTH, VIDEO_HEIGHT


def denormalize_rect(normalized):
    return {
        "x": int(normalized["x"] * VIDEO_WIDTH),
        "y": int(normalized["y"] * VIDEO_HEIGHT),
        "width": int(normalized["width"] * VIDEO_WIDTH),
        "height": int(normalized["height"] * VIDEO_HEIGHT),
    }
